sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -2898,4 +2899,33 @@ class HiveDDLSuite
       }
     }
   }
+
+  test("SPARK-33865: Hive DDL with avro should check col name") {
+    withTable("tbl", "t1") {
+      withView("view1") {
+        spark.sql("CREATE TABLE tbl(id long)")
+        spark.sql("INSERT OVERWRITE TABLE tbl VALUES 4")
+        spark.sql("CREATE VIEW view1 AS SELECT id FROM tbl")
+
+
+        val e1 = intercept[SchemaParseException] {
+          spark.sql(s"CREATE TABLE t1 STORED AS AVRO " +
```

Comment:
Can we also test CREATE TABL without AS SELECT?

Reply:
> Can we also test CREATE TABL without AS SELECT?

Without `as select`, col name is valid since we have check in parser level